aten/src/ATen/core/jit_type.h

```diff
@@ -1105,6 +1105,7 @@ using NumberTypePtr = std::shared_ptr<NumberType>;
 // Subtype hierarchy for Number Types (NumberType as the base type):
 // IntType <: NumberType
 // FloatType <: NumberType
+// ComplexDoubleType <:NumberType
```

Comment:
I know I suggested this initially, but I discussed this with the JIT team and we concluded that it would be better to make `ComplexDoubleType` a subtype of only `AnyType`. `NumberType` is an internal trick used to make sure operators written in Python that take both `int` and `float` (e.g. `norm`, that takes an `int` or `float` for the type of norm to compute) are JITtable. `ComplexDoubleType` cannot be passed in for every parameter that is annotated `NumberType` today.

Reply:
As discussed offline, both actually make sense. Let's follow up on this and fix it later.